src/components/Modal.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+const Modal = ({ handleClose, show, title, body, onAction }) => {
+  return (
+    <div className={show ? "modal display-block" : "modal display-none"}>
+      <div className="modal-main" onClick={handleClose}>
+        <button className="button-div modal-top">
+          <img
+            src="/images/x-icon.svg"
+            alt="close modal icon"
+            className="closeModal-icon"
+          />
+        </button>
+        <div className="modal-main-title">
+          <h1>{title}</h1>
+        </div>
+        <div className="modal-main-content">
+          <p>{body}</p>
+        </div>
+        <div className="modal-main-buttons">
+          <button
+            className="button button-modal button-close"
+            onClick={handleClose}
+          >
+            Cancel
```

Comment:
I like your idea to make the button text for both buttons dynamic. Maybe the props can be optional, and you can fall back to "Cancel" and "Ok" if the props are not provided.